src/__tests__/client.ts

```diff
@@ -450,269 +439,239 @@ describe("client", () => {
+      result: { data },
+    });
+
+    const initialState: any = {
+      data: {
+        'ROOT_QUERY.allPeople({"first":"1"}).people.0': {
+          name: "Luke Skywalker",
+        },
+        'ROOT_QUERY.allPeople({"first":1})': {
+          people: [
+            {
               type: "id",
-              id: 'ROOT_QUERY.allPeople({"first":1})',
               generated: true,
+              id: 'ROOT_QUERY.allPeople({"first":"1"}).people.0',
             },
+          ],
+        },
+        ROOT_QUERY: {
+          'allPeople({"first":1})': {
+            type: "id",
+            id: 'ROOT_QUERY.allPeople({"first":1})',
+            generated: true,
           },
-          optimistic: [],
         },
-      };
-
-      const client = new ApolloClient({
-        link,
-        cache: new InMemoryCache({ addTypename: false }).restore(
-          initialState.data
-        ),
-      });
+        optimistic: [],
+      },
+    };
 
-      expect(client.restore(initialState.data)).toEqual(
-        client.cache.restore(initialState.data)
-      );
+    const client = new ApolloClient({
+      link,
+      cache: new InMemoryCache({ addTypename: false }).restore(
+        initialState.data
+      ),
+    });
 
-      resolve();
-    }
-  );
+    expect(client.restore(initialState.data)).toEqual(
+      client.cache.restore(initialState.data)
+    );
+  });
 
-  itAsync(
-    "should return errors correctly for a single query",
-    (resolve, reject) => {
-      const query = gql`
-        query people {
-          allPeople(first: 1) {
-            people {
-              name
-            }
+  it("should return errors correctly for a single query", async () => {
+    const query = gql`
+      query people {
+        allPeople(first: 1) {
+          people {
+            name
           }
         }
-      `;
+      }
+    `;
 
-      const errors: GraphQLError[] = [
-        new GraphQLError(
-          "Syntax Error GraphQL request (8:9) Expected Name, found EOF"
-        ),
-      ];
+    const errors: GraphQLError[] = [
+      new GraphQLError(
+        "Syntax Error GraphQL request (8:9) Expected Name, found EOF"
+      ),
+    ];
 
-      const link = mockSingleLink({
-        request: { query },
-        result: { errors },
-      }).setOnError(reject);
+    const link = mockSingleLink({
+      request: { query },
+      result: { errors },
+    });
 
-      const client = new ApolloClient({
-        link,
-        cache: new InMemoryCache({ addTypename: false }),
-      });
+    const client = new ApolloClient({
+      link,
+      cache: new InMemoryCache({ addTypename: false }),
+    });
 
-      return client
-        .query({ query })
-        .catch((error: ApolloError) => {
-          expect(error.graphQLErrors).toEqual(errors);
-        })
-        .then(resolve, reject);
-    }
-  );
+    await client.query({ query }).catch((error: ApolloError) => {
+      expect(error.graphQLErrors).toEqual(errors);
+    });
```

Comment:
```suggestion
    await expect(client.query({ query })).rejects.toEqual(
      expect.objectContaining({
        graphQLErrors: errors,
      })
    );
```

As this is currently written it could also resolve without failing

Reply:
Good call!